lib/linux_admin/ssh_agent.rb

```diff
@@ -0,0 +1,85 @@
+require 'net/ssh'
+require 'fileutils'
+module LinuxAdmin
+  class SshAgent
+    attr_reader :pid
+    attr_reader :sock
+
+    def initialize(agent_socket, ssh_private_key)
+      FileUtils.mkdir_p(File.dirname(agent_socket))
+      agent_details = `ssh-agent -a #{agent_socket}`
```

Comment:
I would think so. We would make the private key and (optional) socket the params and store them in instance variables for use in `#start`